### PR TITLE
OS X gitignore cleanup and cocoapods integration tests

### PR DESCRIPTION
**Pull request by thomasvl**
Working toward better cocoapods builds (https://github.com/google/protobuf/issues/1457), this is some initial work to make it easier to test the followup work.

### CODE REVIEW

**Comment by thomasvl** [comment]
Not all of the test pass yet (https://github.com/google/protobuf/issues/1457 cover those issues).

File: objectivec/Tests/CocoaPods/OSXCocoaPodsTester/OSXCocoaPodsTester.xcodeproj/project.xcworkspace/contents.xcworkspacedata [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (perotinus) [inline objectivec/Tests/CocoaPods/OSXCocoaPodsTester/OSXCocoaPodsTester.xcodeproj/project.xcworkspace/contents.xcworkspacedata]: Did you mean to include this file?

File: objectivec/Tests/CocoaPods/OSXCocoaPodsTester/OSXCocoaPodsTester.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -0,0 +1,290 @@
+// !$*UTF8*$!
```
Comment (perotinus) [inline objectivec/Tests/CocoaPods/OSXCocoaPodsTester/OSXCocoaPodsTester.xcodeproj/project.pbxproj]: Can you add a README or note somewhere that says that these are generic, Xcode-generated projects without modifications?

File: objectivec/Tests/CocoaPods/run_tests.sh [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+set -eu
+
+readonly ScriptDir=$(dirname "$(echo $0 | sed -e "s,^\([^/]\),$(pwd)/\1,")")
+
+printUsage() {
+  NAME=$(basename "${0}")
+  cat << EOF
+usage: ${NAME} [OPTIONS]
+
+This script does the common build steps needed.
```
Comment (perotinus) [inline objectivec/Tests/CocoaPods/run_tests.sh]: Should this be more specific? "Runs a suite of build tests with CocoaPods to validate the Protobuf podspec", or something?

File: objectivec/Tests/CocoaPods/run_tests.sh [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      printUsage
+      exit 1
+      ;;
+  esac
+  shift
+done
+
+# Sanity check.
+if [[ "${#TEST_NAMES[@]}" == 0 ]] ; then
+  echo "ERROR: Need to run atleast iOS or OS X tests." 1>&2
```
Comment (perotinus) [inline objectivec/Tests/CocoaPods/run_tests.sh]: at least

File: objectivec/Tests/CocoaPods/OSXCocoaPodsTester/OSXCocoaPodsTester.xcodeproj/project.xcworkspace/contents.xcworkspacedata (hunk 1 above)
Comment (thomasvl) [inline objectivec/Tests/CocoaPods/OSXCocoaPodsTester/OSXCocoaPodsTester.xcodeproj/project.xcworkspace/contents.xcworkspacedata]: Xcode created it, so I left it as is.

File: objectivec/Tests/CocoaPods/OSXCocoaPodsTester/OSXCocoaPodsTester.xcodeproj/project.pbxproj (hunk 2 above)
Comment (thomasvl) [inline objectivec/Tests/CocoaPods/OSXCocoaPodsTester/OSXCocoaPodsTester.xcodeproj/project.pbxproj]: Done.

File: objectivec/Tests/CocoaPods/run_tests.sh (hunk 3 above)
Comment (thomasvl) [inline objectivec/Tests/CocoaPods/run_tests.sh]: done.

File: objectivec/Tests/CocoaPods/run_tests.sh (hunk 4 above)
Comment (thomasvl) [inline objectivec/Tests/CocoaPods/run_tests.sh]: done

File: objectivec/Tests/CocoaPods/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+CocoaPods Protocol Buffers Integration Tests
+============================================
+
+The sub directories are the basic projects as created by Xcode 6.3. They are
+used to then drive `pod` and `xcodebuild` to ensure things integration/build
```
Comment (perotinus) [inline objectivec/Tests/CocoaPods/README.md]: integrate

File: objectivec/Tests/CocoaPods/README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+as expected.
+
+`run_tests.sh` defaults to running all the tests, invoke it with `--help` to
+see the argument to control what tests are run.
```
Comment (perotinus) [inline objectivec/Tests/CocoaPods/README.md]: arguments

**Comment by perotinus** [comment]
Just a few nits, otherwise 👍 .

File: objectivec/Tests/CocoaPods/README.md (hunk 5 above)
Comment (thomasvl) [inline objectivec/Tests/CocoaPods/README.md]: done

File: objectivec/Tests/CocoaPods/README.md (hunk 6 above)
Comment (thomasvl) [inline objectivec/Tests/CocoaPods/README.md]: done

**Pull request merged**